addons/hr_expense/static/src/views/attachment_view.xml

```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<templates xml:space="preserve">
+    <t t-inherit="mail.AttachmentView" t-inherit-mode="extension">
+        <xpath expr="//h3/div/t" position="before">
+            <span t-if="state.thread.model==='hr.expense.sheet'">
+                <t t-out="state.thread.mainAttachment.res_name or state.thread.name" />
```

Comment:
Note that for XML files rendered on the JavaScript side, for echoing text, `t-esc` gives better performance than `t-out`. No big deal though, unless this is a hot path